pkg/core/construct_graph.go

```diff
@@ -12,67 +12,74 @@ import (
 
 type (
 	ConstructGraph struct {
-		underlying *graph.Directed[Construct]
+		underlying *graph.Directed[BaseConstruct]
```

Comment:
so do we need the construct graph now? 

When would something be put in here instead of the resource graph?

Reply:
We don't need it, but it's a smaller diff (code-wise and mentally) to keep it. Given that we don't yet know what's going to replace it, I figured I'd rock the boat as little as possible.

Reply:
thats also an implementation aspect we are going to have to deal with because BaseConstructs can be cyclic and also can exist within the resource graph. It likely isnt an issue as we can just ensure that post expansion theres no cycles but we will have to allow cycles while constructs still exist within the resource graph/dag

Reply:
wait wouldnt they both need to be of BaseConstruct?